oauth2-useragent-core/src/main/java/com/microsoft/alm/oauth2/useragent/AuthorizationResponse.java

```diff
@@ -50,19 +50,27 @@ public String getState() {
     }
 
     public static AuthorizationResponse fromString(final String s) throws AuthorizationException {
+        return fromString(s, null);
+    }
+
+    public static AuthorizationResponse fromString(final String s, final String potentialDescription) throws AuthorizationException {
         String code = null;
         String state = null;
         String error = "unknown_error";
-        String errorDescription = null;
+        String errorDescription = potentialDescription;
         String errorUriString = null;
-        if (s != null) {
+        if (s != null && s.length() > 0) {
             final String[] pairs = PAIR_SEPARATOR.split(s.trim());
 
             for (final String pair : pairs) {
                 final String[] nameAndValue = NAME_VALUE_SEPARATOR.split(pair, 2);
                 try {
                     if (nameAndValue.length != 2) {
-                        throw new AuthorizationException("parsing_error", "Failed to parse server response", null, null);
+                        final StringBuilder sb = new StringBuilder("Failed to parse server response");
+                        if (errorDescription != null) {
+                            sb.append(".\nDetails: ").append(errorDescription);

```

Comment:
> which is a legit server failure.

Hmmm... Yes and no.  I had forgotten that the response being parsed here could come from the remote OAuth 2.0 endpoint as well as from the sub-process that was proxying said request.

The message is probably overly precise here: it's a parsing error, regardless of where that string came from.  In both cases, however, it seems prudent to include the contents of `s` that could not be parsed, in case it could help in troubleshooting/debugging.

As for `stderr`, I expect errors there, by definition.  We could change the wording to not imply so strongly that the contents of `stderr` directly explain those of `stdout`.  What do you think?